Airflow prod (#216)
knowledgesystems/knowledgesystems-k8s-deployment: digits-eks/eks-prod/shared-services/airflow/README.md

# CDSI Airflow

> This folder contains the deployment files for the CDSI Airflow server. For more information, please see this Confluence document: https://mskconfluence.mskcc.org/display/CDSI/Airflow

Reviewer: Is this document repeating some of the content we already have listed in Confluence? In that case, should we avoid the duplication, or do you think it's worth repeating here?

Author: The confluence is more specific and honestly too detailed if all you're trying to do is set up a new instance. These are the instructions that are needed to set up a new instance, and Avery wanted me to put a README here describing those steps.

> Ingress files live separately at [`../ingress`](../ingress).

Reviewer: Maybe link out to the other ingress folder if you plan on keeping that. Mention it's an example.

    4. [Create a storage class for EFS](https://github.com/kubernetes-sigs/aws-efs-csi-driver/blob/master/examples/kubernetes/dynamic_provisioning/README.md).  

4. Allocate persistent volume storage.  
This deployment requires persistent storage for GitHub repo access. If your deployment does not require persistent volume storage, skip this step and remove references to persistent volumes in `override-values.yaml`.  
Before running the below commands, update the `spec.csi.volumeHandle` setting in `pv.yaml` to the filesystem ID from the previous step.  
    ```
    kubectl apply -f storage/pv.yaml -n $NAMESPACE
    kubectl apply -f storage/pvc.yaml -n $NAMESPACE
    ```
	
5. Create secrets.  
The below secrets are required for running the NCI and CDM DAGs. The referenced secrets file is located in the portal-configuration repository.
	```
    kubectl apply -f portal-configuration/k8s-config-vars/digits-eks/eks-prod/airflow-secrets.yaml
    ```

6. Set up networking.  

> In the AWS Route 53 Dashboard, create an entry under the desired hosted zone (In this case, `cbioportal.aws.mskcc.org`):

Reviewer: Perhaps make it clearer that these instructions are specific to development. This domain is the dev domain, in other cases it may not be just a different domain name but also a different provider. ie, the `*.cbioportal.org` sites are managed by Squarespace instead of Route 53.

Author: This domain isn't in dev - it's in the private prod AWS account

> In the AWS Route 53 Dashboard, create an entry under the desired hosted zone (In this case, `cbioportal.aws.mskcc.org`):

Reviewer: ```suggestion
In the AWS Route 53 Dashboard, create a DNS entry under the desired hosted zone (for development purposes, `cbioportal.aws.mskcc.org`):
```

Author: ^ noted above this isn't dev

    - For `Record Name`, enter the desired URL.
    - For `Record Type`, choose `CNAME`. 
    - For `Value`, enter the URL of the desired load balancer.  

7. Apply the ingress.  
Before running the below command, modify `eks-airflow-ingress.yaml` to point to the correct URL for your Airflow server.
    ```
	kubectl apply -f ../ingress/eks-airflow-ingress.yaml -n $NAMESPACE
    ```

8. Modify `webserver.base_url` in `override-values.yaml` to reflect the desired URL of the Airflow server. This is required for email notifications to work correctly.
    ```
    webserver:
        base_url: <airflow-server-url>
    ```

9. Install the Airflow Helm chart.
	```
    helm repo add apache-airflow https://airflow.apache.org
	helm repo update
    helm install -f override-values.yaml airflow apache-airflow/airflow --namespace $NAMESPACE --debug --timeout 10m
    ```

> 10. Setup the DataDog agent.  

Reviewer: Is this something that should go before installing Airflow's Helm chart (ie. so we have better insight into what's happening as the pods are starting up), or does it have to be done afterwards?

Author: I don't think it's necessary to set up datadog before the chart exists

> We monitor our Airflow deployment using MSK's [DataDog instance](https://app.datadoghq.com/account/login/id/861kel1ds8quzqlv

Reviewer: I'm getting an error when I try to visit this page. What is it supposed to link to?

Author: The MSK datadog instance

Reviewer: https://app.datadoghq.com/apm/home
